### DISCUSSION TITLE
Built-in attribute onClose of p-messages

### DISCUSSION THREAD

**Discussion by hudy93** [primefaces/community#1325] [PrimeNG] [answered]
Current behavior:

Unfortunately, as of the current version of PrimeNG, the p-messages component does not provide a built-in way to handle the close event. The p-toast component does provide an onClose event, but it has a different style and behavior.

Expected behavior:

OnClose event is fired if specific message is closed by the user.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

Thanks for the feedback, I've added it to 17.12.0 milestone https://github.com/primefaces/primeng/issues/14982
